client/state/shortcodes/test/actions.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import nock from 'nock';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SHORTCODE_FETCH,
+	SHORTCODE_RECEIVE
+} from 'state/action-types';
+import { fetchShortcode } from '../actions';
+import { useSandbox } from 'test/helpers/use-sinon';
+import wpcom from 'lib/wp';
+
+describe( 'actions', () => {
+	let sandbox, spy;
+
+	useSandbox( newSandbox => {
+		sandbox = newSandbox;
+		spy = sandbox.spy();
+	} );
+
+	describe( '#fetchShortcode()', () => {
+		const siteId = 12345678;
+		const shortcode = '[gallery ids="1,2,3"]';
+
+		describe( 'success', () => {
+			before( () => {
+				nock( 'https://public-api.wordpress.com:443' )
+					.persist()
+					.get( '/sites/' + siteId + '/shortcodes/render' )
+					.query( {
+						shortcode
+					} )
+					.reply( 200, {
+						body: '<html></html>',
+						scripts: [],
+						styles: [],
+						status: 'success'
+					}, {
+						'Content-Type': 'application/json'
+					} );
+			} );
+
+			after( () => {
+				nock.cleanAll();
+			} );
+
+			it( 'should return a fetch action object when called', () => {
+				fetchShortcode( siteId, shortcode )( spy );
+
+				expect( spy ).to.have.been.calledWith( {
+					type: SHORTCODE_FETCH,
+					siteId,
+					shortcode
+				} );
+			} );
+
+			it( 'should return a receive action when request successfully completes', () => {
+				return wpcom.undocumented().site( siteId ).shortcodes(
+					{
+						shortcode
+					},
+					() => {
+						expect( spy ).to.have.been.calledWith( {
+							type: SHORTCODE_RECEIVE,
+							siteId,
+							shortcode,
+							data: {
+								body: '<html></html>',
+								scripts: [],
+								styles: [],
+								status: 'success'
+							},
+							error: null
+						} );
+					}
+				);
+			} );
+		} );
+
+		describe( 'failure', () => {
+			before( () => {
+				nock( 'https://public-api.wordpress.com:443' )
+					.persist()
+					.get( '/sites/' + siteId + '/shortcodes/render' )
+					.query( {
+						shortcode
+					} )
+					.reply( 400, {
+						error: 'The requested shortcode does not exist.'
+					}, {
+						'Content-Type': 'application/json'
+					} );
+			} );
+
+			after( () => {
```

Comment:
The [`useNock` test helper](https://github.com/Automattic/wp-calypso/tree/master/test/test/helpers/use-nock) can help avoid the need for nock cleanup. See also #7875 where I update existing usage.